sunpycube/cube/NDCube.py

```diff
@@ -0,0 +1,238 @@
+            passing values from FITS header or map attributes.
+            See `~astropy.wcs.WCS.wcs_pix2world` for more information.
+
+        quantity_axis_list : `list`
+            A list of `~astropy.units.Quantity`.
+
+        Returns
+        -------
+
+        coord : `list`
+            A list of arrays containing the output coordinates.
+
+        """
+        list_arg = []
+        indexed_not_as_one = []
+        result = []
+        quantity_index = 0
+        for i, _ in enumerate(self.missing_axis):
+            # the cases where the wcs dimension was made 1 and the missing_axis is True
+            if self.missing_axis[self.wcs.naxis-1-i]:
+                list_arg.append(self.wcs.wcs.crpix[i]-1+origin)
+            else:
+                # else it is not the case where the dimension of wcs is 1.
+                list_arg.append(quantity_axis_list[quantity_index])
+                quantity_index += 1
+            # appending all the indexes to be returned in the answer
+                indexed_not_as_one.append(i)
+        pixel_to_world = self.wcs.all_pix2world(*list_arg, origin)
+        # collecting all the needed answer in this list.
+        for index in indexed_not_as_one:
+            result.append(u.Quantity(pixel_to_world[index], unit=self.wcs.wcs.cunit[index]))
+        return result
+
+    def world_to_pixel(self, quantity_axis_list, origin=0):
+        list_arg = []
+        indexed_not_as_one = []
+        result = []
+        quantity_index = 0
+        for i, _ in enumerate(self.missing_axis):
+            # the cases where the wcs dimension was made 1 and the missing_axis is True
+            if self.missing_axis[self.wcs.naxis-1-i]:
+                list_arg.append(self.wcs.wcs.crval[i]+1-origin)
+            else:
+                # else it is not the case where the dimension of wcs is 1.
+                list_arg.append(quantity_axis_list[quantity_index])
+                quantity_index += 1
+            # appending all the indexes to be returned in the answer
+                indexed_not_as_one.append(i)
+        world_to_pixel = self.wcs.all_world2pix(*list_arg, origin)
+        # collecting all the needed answer in this list.
+        for index in indexed_not_as_one:
+            result.append(u.Quantity(world_to_pixel[index], unit=self.wcs.wcs.cunit[index]))
+        return result
+
+    def to_sunpy(self):
+        pass
+
+    @property
+    def dimensions(self):
+        """
+        The dimensions of the data (x axis first, y axis second, z axis third ...so on) and the type of axes.
+        """
+        ctype = list(self.wcs.wcs.ctype)[::-1]
+        axes_ctype = []
+        for i, axis in enumerate(self.missing_axis):
+            if not axis:
+                axes_ctype.append(ctype[i])
+        shape = self.data.shape
+        return PixelPair(dimensions=shape, axes=axes_ctype)
+
+    def plot(self, axes=None, axis_data=['x', 'y'], unit=None, origin=0, *args, **kwargs):
```

Comment:
```axis_data``` should be replaced by ```images_axes``` with the same API as used for ```ImageAnimatorWCS```.  If the Cube is 2D then ```axis_data``` should derived from ```image_axis```.  That way 2D and >2D cubes have the same ploy API.